app/services/rate_limit.rb

```diff
@@ -1,24 +1,24 @@
 # This class is similar to RedisRateLimiter, but differs in that
 # the throttle period begins once the maximum number of allowed
 # attempts has been reached.
-class Throttle
-  attr_reader :throttle_type
+class RateLimit
```

Comment:
WDYT of renaming this to `RateLimiter`? I like to name service classes to indicate the verbs they do, because we're not instantiating a rate limit itself here, we're instantiating a thing that helps enforce rate limits?